MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -1,72 +1,122 @@
+    
+    func didRecieveNextQuestion(question: QuizQuestion?) {
+        guard let question = question else {
+            return
+        }
+        
+        currentQuestion = question
+        let viewModel = convert(model: question)
+        DispatchQueue.main.async { [weak self] in
+            self?.show(quiz: viewModel)
+        }
+    }
+    
+    // MARK: - AlertPresenterDelegate
+    
+    func didShow(alert: UIAlertController) {
+        self.present(alert, animated: true)
+    }
+    
+    @IBAction private func yesButtonClicked(_ sender: UIButton) {
+        guard let currentQuestion = currentQuestion else { return }
+        let givenAnswer = true
+        sender.isEnabled = false
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
+    }
+    
+    @IBAction private func noButtonClicked(_ sender: UIButton) {
+        guard let currentQuestion = currentQuestion else { return }
+        let givenAnswer = false
+        sender.isEnabled = false
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
+    }
+    
+    private func show(quiz step: QuizStepViewModel) {
+        // заполняем нашу картинку, текст и счётчик данными
+        imageView.image = step.image
+        textLabel.text = step.question
+        counterLabel.text = step.questionNumber
+    }
+    
+    private func convert(model: QuizQuestion) -> QuizStepViewModel {
+        return QuizStepViewModel(
+            image: UIImage(named: model.image) ?? UIImage(),
+            question: model.text,
+            questionNumber: "\(currentQuestionIndex + 1)/\(questionsAmount)"
+        )
+    }
+    
+    private func showAnswerResult(isCorrect: Bool) {
+        if isCorrect {
+            correctAnswers += 1
+        }
+        
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderWidth = 8
+        imageView.layer.borderColor = isCorrect ? UIColor.ypGreen.cgColor : UIColor.ypRed.cgColor
+        
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [weak self] in
+            guard let self = self else { return }
+            self.showNextQuestionOrResults()
+            self.imageView.layer.borderWidth = 0
+            self.noButton.isEnabled = true
+            self.yesButton.isEnabled = true
+        }
+    }
+    
+    private func showNextQuestionOrResults() {
+        if currentQuestionIndex == questionsAmount - 1 {
+            statisticService.store(correct: correctAnswers, total: questionsAmount)
+            let totalAccuracy = statisticService.totalAccuracy
+            let gamesCount = statisticService.gamesCount
+            let bestGames = statisticService.bestGame
+            let text = """
+Ваш результат \(correctAnswers)/\(questionsAmount)
+Количество сыгранных квизов: \(gamesCount)
+Рекорд: \(bestGames.correct)/\(bestGames.total) (\(bestGames.date.dateTimeString))
+Средняя точность: \(String(format: "%.2f", totalAccuracy))%
+"""
```

Comment:
Хорошо, когда код соблюдает столбчатость.  К сожалению с такими строками автоформаттер не умеет у xcode работать :( 
```suggestion
            Ваш результат \(correctAnswers)/\(questionsAmount)
            Количество сыгранных квизов: \(gamesCount)
            Рекорд: \(bestGames.correct)/\(bestGames.total) (\(bestGames.date.dateTimeString))
            Средняя точность: \(String(format: "%.2f", totalAccuracy))%
            """
```

Reply:
Если строки сдвинуть, то на алерте они сместятся, поэтому невозможно отредактировать в столбец

Reply:
вообще не должны, но надо посмотреть

Reply:
@etdevios 
<img width="1197" alt="Снимок экрана 2022-12-01 в 20 26 44" src="https://user-images.githubusercontent.com/20269264/205131554-842125b4-a493-4c1e-85ac-71f48670e4a6.png">

Reply:
я пробывал, они уезжали.... видимо, что-то пошло не так

Reply:
да, важно чтобы три кавычки тоже стояли в том же столбце.

Reply:
поэксперементируй)

Reply:
Точно! Когда писал эти строки не обратил внимание где были кавычки, остальной текст смещал) Буду знать, спасибо еще раз!